client/my-sites/domains/domain-management/name-servers/with-domain-nameservers.js

```diff
@@ -19,57 +19,26 @@ const noticeOptions = {
 	id: `nameserver-update-notification`,
 };
 
-const useSuccessNotice = ( isSuccess ) => {
-	const showNotice = useRef();
-	const dispatch = useDispatch();
-	const translate = useTranslate();
-
-	useEffect( () => {
-		showNotice.current = () => {
-			dispatch(
-				successNotice(
-					translate( 'Yay, the name servers have been successfully updated!' ),
-					noticeOptions
-				)
-			);
-		};
-	}, [ dispatch, translate ] );
-
-	useEffect( () => {
-		isSuccess && showNotice.current();
-	}, [ isSuccess ] );
-};
-
-const useErrorNotice = ( isError, error ) => {
-	const showNotice = useRef();
-	const dispatch = useDispatch();
-	const translate = useTranslate();
-
-	useEffect( () => {
-		showNotice.current = () => {
-			const defaultMessage = translate( 'An error occurred while updating the nameservers.' );
-			dispatch( errorNotice( error?.message ?? defaultMessage, noticeOptions ) );
-		};
-	}, [ dispatch, translate, error ] );
-
-	useEffect( () => {
-		isError && showNotice.current();
-	}, [ isError ] );
-};
-
 const withDomainNameservers = createHigherOrderComponent( ( Wrapped ) => {
 	const WithDomainNameservers = ( props ) => {
 		const { selectedDomainName } = props;
+		const dispatch = useDispatch();
+		const translate = useTranslate();
 		const { data, isLoading, isError, error } = useDomainNameserversQuery( selectedDomainName );
-		const {
-			updateNameservers,
-			isSuccess: isUpdateSuccess,
-			isError: isUpdateError,
-			error: updateError,
-		} = useUpdateNameserversMutation( selectedDomainName );
-
-		useSuccessNotice( isUpdateSuccess );
-		useErrorNotice( isUpdateError, updateError );
+		const { updateNameservers } = useUpdateNameserversMutation( selectedDomainName, {
+			onSuccess() {
+				dispatch(
+					successNotice(
+						translate( 'Yay, the name servers have been successfully updated!' ),
+						noticeOptions
+					)
+				);
+			},
+			onError( err ) {
+				const defaultMessage = translate( 'An error occurred while updating the nameservers.' );
+				dispatch( errorNotice( err?.message ?? defaultMessage, noticeOptions ) );
```

Comment:
A better practice would be to inline the `translate()` call:
```js
err?.message ?? translate( ... )
```
That way the `defaultMessage` is constructed only when really needed.

Reply:
good call! I'll make sure to change that on the parent PR https://github.com/Automattic/wp-calypso/pull/53657